### PR TITLE
feat(Grid): Convert grid to typescript

### LINKED ISSUE #2447: Convert Grid to Typescript
Convert grid to typescript, add demo, and integration testing

### PR DESCRIPTION
**Pull request by jessiehuff**
Fixes #2447

### CODE REVIEW

File: packages/patternfly-4/react-core/src/layouts/Grid/Grid.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+export type gridItemSpanValueShape = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
+
+export interface GridProps extends React.HTMLProps<HTMLDivElement> {
+  /** content rendered inside the Grid layout */
+  children?: React.ReactNode; 
+  /** additional classes added to the Grid layout */
+  className?: string; 
+  /** Adds space between children. Options are sm, md or lg */
+  gutter?: boolean;
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/layouts/Grid/Grid.tsx]: Same comment here about changing the type of gutter.  Why are we doing that?

File: packages/patternfly-4/react-core/src/components/Title/Title.test.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { Title, TitleSize, TitleLevel } from '.';
 Object.values(TitleSize).forEach(size => {
   test(`${size} Title`, () => {
     const view = shallow(
-      <Title size={size} headingLevel={TitleLevel.h1}>
+      <Title size={size as 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl'} headingLevel={TitleLevel.h1}>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Title/Title.test.tsx]: does the `Title.h1` work here?  Or do we have to use the string?

File: packages/patternfly-4/react-core/src/styles/sizes.ts [hunk 3]
```diff
@@ -10,9 +10,9 @@ export const BaseSizes = {
...
-  [BaseSizes.sm]: BaseSizes.sm,
-  [BaseSizes.md]: BaseSizes.md,
-  [BaseSizes.lg]: BaseSizes.lg,
-  [BaseSizes.xl]: BaseSizes.xl,
-  xl2: BaseSizes['2xl']
+  sm: 'Sm',
+  md: 'Md',
+  lg: 'Lg',
+  xl: 'Xl',
+  xl2: '_2xl'
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/styles/sizes.ts]: Why did we change the definition of the const here?

Comment (jessiehuff) [reply packages/patternfly-4/react-core/src/styles/sizes.ts]: This is to build the prop names from react-styles correctly.

File: packages/patternfly-4/react-core/src/components/Title/Title.test.tsx (hunk 2 above)
Comment (jessiehuff) [reply packages/patternfly-4/react-core/src/components/Title/Title.test.tsx]: Yes, it still works. :)

File: packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.test.tsx [hunk 4]
```diff
@@ -7,7 +7,6 @@ import { EmptyStateSecondaryActions } from './EmptyStateSecondaryActions';
 import { EmptyStateIcon } from './EmptyStateIcon';
 import { Button } from '../Button';
 import { Title } from '../Title';
-import {BaseSizes} from "../../styles/sizes";
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/EmptyState/EmptyState.test.tsx]: why did you change this file?  Can we open a separate PR?

File: packages/patternfly-4/react-core/src/components/Popover/PopoverHeader.tsx [hunk 5]
```diff
@@ -3,7 +3,7 @@ import { Title, TitleSize } from '../Title';
 import { Omit } from '../../helpers/typeUtils';
 
 export const PopoverHeader: React.FunctionComponent<PopoverHeaderProps> = ({ children, id, ...props }) => (
-  <Title headingLevel="h6" size={TitleSize.xl} id={id} {...props}>
```
Comment (tlabaj) [inline packages/patternfly-4/react-core/src/components/Popover/PopoverHeader.tsx]: why did popover change.?

File: packages/patternfly-4/react-core/src/components/Title/Title.test.tsx (hunk 2 above)
Comment (tlabaj) [reply packages/patternfly-4/react-core/src/components/Title/Title.test.tsx]: Okay, butwhy are we changing this as part of the Grid conversion?

Comment (jessiehuff) [reply packages/patternfly-4/react-core/src/components/Title/Title.test.tsx]: I was running into some errors with the tests, and this was my workaround at the time. I think I found a way that we can keep the previous code.
